### PR TITLE
docs: add CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by sploiselle**
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/4980)

### CODE REVIEW

File: doc/user/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+This doc is a WIP. There is more useful information in the [README](./README.md).
+
+# Set up
+
+Rather than work on `github.com/materializeinc/materialize` itself, we recommend
+working on your own fork of Materialize; this makes branches much tidier.
+branches tidier.
+
+1. Fork `github.com/materializeinc/materialize`. You can do this by going to
+   that address and clicking **Fork**.
```
Comment (quodlibetor) [inline doc/user/CONTRIBUTING.md:10]: this could plausibly be a link instead of preventing linking by putting it in code?

```suggestion
1. Fork <https://github.com/materializeinc/materialize>. You can do this by going to
   that address and clicking **Fork**.
```

File: doc/user/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+1. Set `materializeinc/materialize` as your `upstream` remote:
+
+    ```shell
+    git remote add upstream git@github.com:MaterializeInc/materialize.git
+    ```
+
+This will give you your own workspace to use while still giving you the ability
+to `git pull` the latest version of Materialize.
+
```
Comment (quodlibetor) [inline doc/user/CONTRIBUTING.md:32]: I think it's worth linking to `../developer/guide.md` for detailed work instructions. It's fairly long, so I personally would caveat that folks don't need to read the whole thing if they just want to issue a drive-by PR, maybe something like:

```markdown
We follow a standard github pull-request workflow. For details on our standards and workflow please consult [the developer guide](../developer/guide.md).
```

**Review by quodlibetor — APPROVED** [review]
LGTM, small linkifications.

File: doc/user/CONTRIBUTING.md (hunk 2 above)
Comment (sploiselle) [reply doc/user/CONTRIBUTING.md:32]: Ah, this is for contributing to the docs; will make that clearer.

Comment (quodlibetor) [reply doc/user/CONTRIBUTING.md:32]: oh, lol, that makes sense! I *was* confused.

File: doc/user/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
+This doc is a WIP. There is more useful information in the [README](./README.md).
+
+# Set up
+
+Rather than work on `github.com/materializeinc/materialize` itself, we recommend
+working on your own fork of Materialize; this makes branches much tidier.
+branches tidier.
```
Comment (andrioni) [inline doc/user/CONTRIBUTING.md:7]: Potential nitpick: only people with write access to this repo can create a branch inside it, all external contributors are forced to fork it.
I guess this depends on the target audience of this document: people at Materialize or contributors in general

File: doc/user/README.md [hunk 4]
```diff
@@ -3,29 +3,7 @@
...
-    ```shell
-    hugo server --disableFastRender --ignoreCache
-    ```
-
-1. Open <http://localhost:1313>.
-
-You can also read the documentation in Markdown in the `content` directory,
-though some features like our SQL syntax diagrams will not be readily
-accessible.
+For help contributing to the docs, see [`CONTRIBUTING.md.md`](./CONTRIBUTING.md).
```
Comment (andrioni) [inline doc/user/README.md:6]:
```suggestion
For help contributing to the docs, see [`CONTRIBUTING.md`](./CONTRIBUTING.md).
```

File: doc/user/README.md [hunk 5]
```diff
@@ -3,29 +3,7 @@
 This is the root directory of the Materialize user documentation, which are
 rendered by [Hugo] locally and published by CI to <https://materialize.com/docs>.
```
Comment (msudberg) [inline doc/user/README.md:4]: "which is", not "which are" (suspect someone changed "docs" to "documentation" at some point.

**Comment by cirego** [comment]
Random question: is this document better suited for the developer docs or the user docs?

**Comment by sploiselle** [comment]
@cirego This is for the user docs, though one could certainly atomize the bits differently to achieve either goal.

**Comment by cirego** [comment]
> @cirego This is for the user docs, though one could certainly atomize the bits differently to achieve either goal.

@sploiselle, I'm not sure you answered my question. Perhaps I should have asked "Why is this in the user docs and not in the developer docs?" Contributing seems like a "developing" thing, not a "using" thing.

**Comment by sploiselle** [comment]
@cirego it's for people contributing to the documentation
